spec/javascript/utils/compiler/variables.test.js

```diff
@@ -397,5 +394,68 @@ describe("variables.js", () => {
           .map((name) => [name, getDefaultVariableNameIndex(name)]))
       ).toStrictEqual(expectedOutput)
     })
+
+    test("Should get Global variables from parameter objects", () => {
+      completionsMap.set(setCompletionsMap("For Global Variable"))
+      const input = "For Global Variable({ One: variable1, Two: Global.variable2, Three: Count Of(Array()), Four: 1 })"
+      const expectedOutput = {
+        globalVariables: ["variable2", "variable1"],
+        playerVariables: []
+      }
+  
+      expect(getVariables(input)).toEqual(expectedOutput)
+    })
+  
+    test("Should get variables from nested parameter objects", () => {
+      completionsMap.set(setCompletionsMap("For Global Variable"))
+      const input =
+        `For Global Variable({
+          One: variable1,
+          Two: 0,
+          Three: Some Action({ 
+            First: Global.variable2,
+            Second: Event Player.variable3
+          }),
+          Four: 1
+        })`
+      const expectedOutput = {
+        globalVariables: ["variable2", "variable1"],
+        playerVariables: ["variable3"]
+      }
+  
+      expect(getVariables(input)).toEqual(expectedOutput)
+    })
+  
+    test("Should get Player variables from parameter objects", () => {
+      completionsMap.set(setCompletionsMap("Chase Player Variable Over Time"))
+      const input =
+        `Chase Player Variable Over Time({ 
+          One: Event Player,
+          Two: variable1,
+          Three: 0,
+          Four: 1,
+          Five: Destination And Duration
+        })`
+      const expectedOutput = {
+        globalVariables: [],
+        playerVariables: ["variable1"]
+      }
+  
+      expect(getVariables(input)).toEqual(expectedOutput)
+    })
   })
 })
+
+function setCompletionsMap(actionLabel) {
+  return [{
+    label: actionLabel,
+    args_length: 4,
+    parameter_keys: ["One", "Two", "Three", "Four"],
+    parameter_defaults: ["A", "B", "C", "D"]
```

Comment:
I'd say lets keep these real.

E.g. for `For Global Variable`:
```js
{
  label: "For Global Variable",
  args_length: 4,
  parameter_keys: ["Control Variable", "Range Start", "Range Stop", "Step"]
  parameter_defaults: ["variableName", "0", "Count Of(Array())", "1"]
}
```

E.g. for `Chase Player Variable Over Time`:
```js
{
  label: "Chase Player Variable Over Time",
  args_length: 5,
  parameter_keys: ["Player", "Variable", "Destination", "Rate", "Reevaluation"]
  parameter_defaults: ["Event Player", "variableName", "0", "1", "Destination And Rate"]
}
```

You can get these from autocompletion in the Editor :)

Reply:
Except `getVariables()` relies heavily in order of parameters, and also it makes it easier to read the test inputs if they match how they look on real Workshop.